File: app/src/main/java/com/werocksta/dagger2demo/di/module/AndroidModule.java
package com.werocksta.dagger2demo.di.module;

import android.support.customtabs.CustomTabsIntent;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidModule {

    @Provides
    @Singleton
    public CustomTabsIntent.Builder provideCustomTabsBuilder() {
        return new CustomTabsIntent.Builder();
    }

    @Provides
    @Singleton
    public CustomTabsIntent provideCustomTabsIntent(CustomTabsIntent.Builder builder) {
        return builder.build();
    }
}

File: app/src/main/java/com/werockstar/dagger2demo/di/module/AndroidModule.kt
package com.werockstar.dagger2demo.di.module

import android.support.customtabs.CustomTabsIntent

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AndroidModule {

    @Provides
    @Singleton
    fun provideCustomTabsBuilder(): CustomTabsIntent.Builder = CustomTabsIntent.Builder()

    @Provides
    @Singleton
    fun provideCustomTabsIntent(builder: CustomTabsIntent.Builder): CustomTabsIntent = builder.build()
    
}
